azure-quantum/tests/unit/test_problem.py

```diff
@@ -346,8 +361,62 @@ def test_terms_from_npz_pubo(self):
             ),
             self.pubo_problem.terms
         )
+    
+    def test_serialzie_proto_problem(self):
+        problem = Problem(name = "test_proto", problem_type = ProblemType.ising, serialization_type="application/x-protobuf")
+        problem.terms = [
+            Term(c=3, indices=[1, 0]),
+            Term(c=5, indices=[2, 0]),
+            Term(c=3, indices=[1, 0]),
+            Term(c=5, indices=[2, 0]),
+            Term(c=3, indices=[1, 0]),
+            Term(c=5, indices=[2, 0]),
+            Term(c=3, indices=[1, 0]),
+            Term(c=5, indices=[2, 0]),
+            Term(c=3, indices=[1, 0]),
+            Term(c=5, indices=[2, 0]),
+            Term(c=3, indices=[1, 0]),
+            Term(c=5, indices=[2, 0]),
+        ]
+        problem_msgs = problem.serialize()
+        self.assertEqual(
+            len(problem_msgs), 1
+        )
+        proto_problem = problem_pb2.Problem()
+        proto_problem.ParseFromString(problem_msgs[0])
+        self.assertEqual(
+            proto_problem.cost_function.type, 
+            problem_pb2.Problem.ProblemType.ISING
+        )
+        self.assertEqual(
+            len(proto_problem.cost_function.terms),
+            12
+        )
+    
+    def test_deserialize_proto_problem(self):
+        problem = Problem(name = "test_proto", problem_type = ProblemType.pubo, serialization_type="application/x-protobuf")
+        problem.terms = [
+            Term(c=3, indices=[1, 0]),
+            Term(c=5, indices=[2, 0]),
+            Term(c=3, indices=[1, 0]),
+            Term(c=5, indices=[2, 0]),
+            Term(c=3, indices=[1, 0]),
+            Term(c=5, indices=[2, 0]),
+            Term(c=3, indices=[1, 0]),
+            Term(c=5, indices=[2, 0]),
+            Term(c=3, indices=[1, 0]),
+            Term(c=5, indices=[2, 0]),
+            Term(c=3, indices=[1, 0]),
+            Term(c=5, indices=[2, 0]),
+        ]
+        problem_msgs = problem.serialize()
+        deserialized_problem = Problem.deserialize(problem_msgs)
+        self.assertEqual( len(deserialized_problem.terms), 12 )
+        self.assertEqual(deserialized_problem.problem_type, ProblemType.pubo)
+        self.assertEqual(deserialized_problem.name, problem.name)
+        #self.assertEqual( len(deserialized_problem.terms), 12 )
 
```

Comment:
Can we add a test that checks that if we upload a problem with the `application/x-protobuf` encoding, it can be deserialized without having to pass the encoding itself?

Reply:
Hi, if the content type is not specified when creating problem, a json problem will be created. 
In this test, the content type protobuf is specified only for creating the problem, the deserialzition function is called without specifying the content type. Is that what you were expecting to test?